### PR TITLE
Delete duplicated tests in iree/test/e2e/

### PR DESCRIPTION
**Pull request by hanhanW**
Also delete unnecessary flags in iree/test/e2e and iree/compiler/Dialect/HAL/Target/LLVM/test/smoketest.mlir

Part of https://github.com/google/iree/issues/5442

### CODE REVIEW

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Lets keep this for now. If you want you can create a new column on Linalg on tensors project, and add this PR to that

**Comment by hanhanW** [comment]
> Lets keep this for now. If you want you can create a new column on Linalg on tensors project, and add this PR to that

The flag is not used. It now looks into the target backend to decide whether dispatch Linalg on tensors. I can make it still usable so you can control the behavior easier.

https://github.com/google/iree/blob/1a941fc4fd4398c83e168281e5cf657c663e8e40/iree/compiler/Translation/IREEVM.cpp#L132-L138

I'd like to delete duplicated and outdated e2e tests at least because it blocks mhlo.fft end-to-end testing. mhlo.fft only works in Linalg on tensors. What do you think?

**Comment by MaheshRavishankar** [comment]
> > Lets keep this for now. If you want you can create a new column on Linalg on tensors project, and add this PR to that
> 
> The flag is not used. It now looks into the target backend to decide whether dispatch Linalg on tensors. I can make it still usable so you can control the behavior easier.
> 
> https://github.com/google/iree/blob/1a941fc4fd4398c83e168281e5cf657c663e8e40/iree/compiler/Translation/IREEVM.cpp#L132-L138
> 
> I'd like to delete duplicated and outdated e2e tests at least because it blocks mhlo.fft end-to-end testing. mhlo.fft only works in Linalg on tensors. What do you think?

I think deleting the tests is fine. We might still need the old flag to run the legacy path for perf comparison for maybe a week or so.

**Comment by hanhanW** [comment]
> > > Lets keep this for now. If you want you can create a new column on Linalg on tensors project, and add this PR to that
> > 
> > 
> > The flag is not used. It now looks into the target backend to decide whether dispatch Linalg on tensors. I can make it still usable so you can control the behavior easier.
> > https://github.com/google/iree/blob/1a941fc4fd4398c83e168281e5cf657c663e8e40/iree/compiler/Translation/IREEVM.cpp#L132-L138
> > 
> > I'd like to delete duplicated and outdated e2e tests at least because it blocks mhlo.fft end-to-end testing. mhlo.fft only works in Linalg on tensors. What do you think?
> 
> I think deleting the tests is fine. We might still need the old flag to run the legacy path for perf comparison for maybe a week or so.

Okay, I will make the flag work and delete the duplicate tests in this PR instead.

File: iree/compiler/Dialect/Flow/Transforms/Passes.cpp [hunk 1]
```diff
@@ -32,6 +32,12 @@
 #include "mlir/Pass/PassRegistry.h"
 #include "mlir/Transforms/Passes.h"
 
+static llvm::cl::opt<bool> clEnableLinalgOnTensorsDispatch(
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/Flow/Transforms/Passes.cpp]: This should be default true?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/test/e2e/vulkan_specific/BUILD [hunk 2]
```diff
@@ -74,9 +74,6 @@ iree_check_single_backend_test_suite(
         "vectorized_conv.mlir",
     ],
     compiler_flags = [
-        "-iree-flow-dispatch-linalg-on-tensors",
-        "-iree-codegen-spirv-experimental-linalg-on-tensors",
-        "-iree-spirv-enable-vectorization",
```
Comment (MaheshRavishankar) [inline iree/test/e2e/vulkan_specific/BUILD:79]: This flag probably needs to be kept (I am talking about just `iree-spirv-enable-vectorization`?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/test/e2e/vulkan_specific/BUILD (hunk 2 above)
Comment (hanhanW) [reply iree/test/e2e/vulkan_specific/BUILD:79]: No, we don't need it today. Vectorization is turned on by default in Linalg on tensors, see https://github.com/google/iree/pull/5280

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**